src/GroupManager.php

```diff
@@ -123,10 +229,39 @@ public function removeGroup($entity_type_id, $bundle_id) {
   }
 
   /**
+   * Refreshes the locally stored data.
+   *
+   * Call this after making a change to a relationship between a group type and
+   * a group content type.
+   */
+  public function refresh() {
```

Comment:
It is a reset for the groupRelationMap, but not for the groupMap.

Shall I expand the magic getter trick for the groupMap too? Then we can make this a pure cache invalidation that unsets the caches. The magic getter will then repopulate on demand.